pkg/controllers/user/endpoints/endpoints.go

```diff
@@ -221,3 +230,39 @@ func getNodeNameToMachine(clusterName string, machineLister managementv3.NodeLis
 	}
 	return machineMap, nil
 }
+
+func convertIngressToServicePublicEndpointsMap(obj *extensionsv1beta1.Ingress) (map[string][]v3.PublicEndpoint, error) {
+	var addresses []string
+	for _, address := range obj.Status.LoadBalancer.Ingress {
+		addresses = append(addresses, address.IP)
+	}
+	joinedAddresses := strings.Join(addresses, ",")
+	epsMap := make(map[string][]v3.PublicEndpoint)
+	for _, rule := range obj.Spec.Rules {
+		for _, path := range rule.HTTP.Paths {
+			p := v3.PublicEndpoint{
+				Hostname:    rule.Host,
+				Path:        path.Path,
+				ServiceName: fmt.Sprintf("%s:%s", obj.Namespace, path.Backend.ServiceName),
+				Address:     joinedAddresses,
```

Comment:
Currently public endpoint structure expects to see a single address in this field. If we decide to support multiple ones, it should be defined as []string in rancher/types as separating addresses by , format should be explained to the user explicitly. Going to follow up with @vincent99 on this one.